linkml/generators/jsongen.py

```diff
@@ -0,0 +1,91 @@
+import logging
+import os
+import json
+from typing import Dict, List, TextIO, Union
+
+import click
+
+from linkml_runtime.dumpers import json_dumper
+
+from linkml_runtime.linkml_model.meta import SchemaDefinition, ClassDefinition
+
+from linkml_runtime.utils.schemaview import SchemaView
+
+from linkml.utils.generator import Generator, shared_arguments
+
+
+logger = logging.getLogger(__name__)
+
+
+class JsonGenerator(Generator):
+    generatorname = os.path.basename(__file__)
+    generatorversion = "1.0.0"
+    valid_formats = ["json"]
+
+    def __init__(
+        self,
+        schema: Union[str, TextIO, SchemaDefinition],
+        class_name: str,
+        **kwargs,
+    ):
+        self.schemaview = SchemaView(schema)
+        self.class_name = class_name
+
+    def _get_class_defn(self, class_name: str) -> ClassDefinition:
```

Comment:
minor stylistic point: it's probably not worth having a method for this, just call schemaview directly. But YMMV

Reply:
Hm, perhaps this will reduce redundancy when we add more options to `gen-linkml`, like the YAML format option?